File: app/src/main/java/me/imzack/app/end/view/contract/TypeEditViewContract.java
package me.imzack.app.end.view.contract;

import android.support.annotation.DrawableRes;

import me.imzack.app.end.model.bean.FormattedType;

public interface TypeEditViewContract extends BaseViewContract {

    void showInitialView(FormattedType formattedType);

    void showTypeNameEditorDialog(String originalEditorText);

    void onTypeNameChanged(String typeName, String firstChar);

    void onTypeMarkColorChanged(int colorInt, String colorName);

    void onTypeMarkPatternChanged(boolean hasPattern, @DrawableRes int patternResId, String patternName);

    void showTypeMarkColorPickerDialog(String defaultColor);

    void showTypeMarkPatternPickerDialog(String defaultPattern);
}

File: app/src/main/kotlin/me/imzack/app/end/view/contract/TypeEditViewContract.kt
package me.imzack.app.end.view.contract

import android.support.annotation.DrawableRes

import me.imzack.app.end.model.bean.FormattedType

interface TypeEditViewContract : BaseViewContract {

    fun showInitialView(formattedType: FormattedType)

    fun showTypeNameEditorDialog(originalEditorText: String)

    fun onTypeNameChanged(typeName: String, firstChar: String)

    fun onTypeMarkColorChanged(colorInt: Int, colorName: String)

    fun onTypeMarkPatternChanged(hasPattern: Boolean, @DrawableRes patternResId: Int, patternName: String?)

    fun showTypeMarkColorPickerDialog(defaultColor: Int)

    fun showTypeMarkPatternPickerDialog(defaultPattern: String?)
}
